src/App.js

```diff
@@ -1,16 +1,45 @@
-import React from 'react';
+import React, {useState} from 'react';
 import './App.css';
+import ChatLog from './components/ChatLog';
+import TimeStamp from './components/TimeStamp';
 import chatMessages from './data/messages.json';
 
+
 const App = () => {
+
+  const [messages, setMessages] = useState(chatMessages);
+
+  const updateMessage = (messageId) => {
+
+    const updatedMessages = messages.map(message => {
+      const newMessage = {...message}
+      if (message.id === messageId) {
+        newMessage.liked = !newMessage.liked;
+      }
+      return newMessage
+    })
+    setMessages(updatedMessages);
+  }
```

Comment:
Nice work updating the state data by returning a brand new object with map and the spread operator!

Here's a refresher on updating state in React:

React components re-render whenever there is a change/update to state data. In this case, we are modifying a key-value pair within an object in a nested data structure.

Under the hood, React uses Object.is() to compare the previous state object with the one that's been provided viasetMessages(updatedMessages). Object.is() checks if the object passed into the set state function has a different reference in memory. In Javascript, changing the properties and/or values in an object does NOT change the object reference, which is why we must create a new version of our state object (contains a copy of all the properties that weren't updated along with the properties/values that were).

Here is an article with more info: https://www.valentinog.com/blog/react-object-is/

Reply:
Nice work using the spread `...` operator ! We can also refactor the following using a ternary:

```javascript
    const updatedMessages = messages.map(message => {
      message.id === messageId ? {...message: liked: !message.liked } : message
    })
    setMessages(updatedMessages);
  }
```